Makefile

```diff
@@ -0,0 +1,52 @@
+MAKEFLAGS += --no-builtin-rules
+MAKEFLAGS += --no-builtin-variables
+
+xdsl_sources = $(shell find xdsl -type f)
+test_sources = $(shell find tests -type f)
+filecheck_sources = $(shell find tests -iname '*.mlir' -type f)
+
+
+.PHONY: clean filecheck pytest tests rerun-notebooks precommit-install precommit black pyright
+
+# remove all caches and the venv
+clean:
+	rm -rf venv .pytest_cache *.egg-info .coverage.*
```

Comment:
Can you make the `venv` defined in a variable (e.g. `VENV_DIR`) having the default value `venv` but allowing a user to change this from the command line.
I don't name my venv always `venv`.

This usually uses the `?=` [operator][1].

[1]: https://www.gnu.org/software/make/manual/html_node/Setting.html

Reply:
It should be ok with a sensible default. Also, what makefile isn't brittle ;)

Reply:
Technically not correct, the gitignore has `.env, .venv, ENV, env, venv, env.bak, venv.bak`